specs/base_directory_specification.md

```diff
@@ -1,224 +1,63 @@
-    non-essential (cached) data should be written. This directory is
-    defined by the environment variable `$XDG_CACHE_HOME`.
-
--   There is a single base directory relative to which user-specific
-    runtime files and other file objects should be placed. This
-    directory is defined by the environment variable `$XDG_RUNTIME_DIR`.
-
-All paths set in these environment variables must be absolute. If an
-implementation encounters a relative path in any of these variables it
-should consider the path invalid and ignore it.
-
-All configuration paths must support drop-ins. Supporting drop-ins means
-that in addition to parsing `$XDG_CONFIG_DIRS/foo/bar`, an implementation
-also must parse all files in `$XDG_CONFIG_DIRS/foo/bar.d/`, and merge the
-result in lexicographic order.
-[libeconf](https://github.com/openSUSE/libeconf) supports drop-ins and
-can be used by any program to load its configuration following this
-specification.
-In addition, it must be possible to mask files across different locations
-in `$XDG_CONFIG_DIRS` by creating a symlink to `/dev/null` or an empty
-file. For example, an empty `/etc/foo` means that `/usr/etc/foo` is
+Various specifications attempt to define configuration files and file formats. This
+specification establishes where these files should be looked for, in which
+order, and how precedence, masking, extensions and overrides work.
+
+The purpose of the rules defined here is to allow OS vendors to implement the
+hermetic-usr pattern, where all vendor files are shipped in the vendor tree itself
+(`/usr/`), including configuration files with system defaults, while allowing local
+or vendor overrides without modifying the original files, for easier management. This is
+especially beneficial for image-based deployments, where the vendor tree is read-only.
+
+These rules are derived from existing real-world usage from the [systemd
+project](https://github.com/systemd/systemd) and the [libeconf
+project](https://github.com/openSUSE/libeconf).
+
+This specification is agnostic toward the actual format and content of the configuration
+files, the precise path used under each top-level hierarchy, and also toward the
+filenames and extensions, with the exception of the `.d/` notation for drop-ins
+directories.
+
+## Storage Directories and Overrides
+In order to allow shipping system defaults owned by the OS vendor, while at the same
+time letting local users or admins override those defaults, `/usr/` and `/etc/` are both
+supported for storage of configuration files, with the latter having higher priority.
+Programs must work correctly if no configuration files are found in `/etc/`.
+Optionally, `/run/` is also supported for ephemeral overrides.
+
+For example, `/usr/foo/bar.conf` provides the default configuration file, but a user
+can create `/etc/foo/bar.conf` which would take precedence and completely override
+the former. If `/run/foo/bar.conf` is present and supported, it would take precedence
+over both.
```

Comment:
This differs from the documented order for [systemd.unit files](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) at least for drop-ins:

> In addition to /etc/systemd/system, the drop-in ".d/" directories for system services can be placed in /usr/lib/systemd/system or /run/systemd/system directories. Drop-in files in /etc/ take precedence over those in /run/ which in turn take precedence over those in /usr/lib/.

The order for unit files is more complex, but `/etc/systemd/system/*` is searched before `/run/systemd/system/*`, which is searched before `/usr/lib/systemd/system/*`. 

libeconf doesn't seem to support `/run`:

- https://opensuse.github.io/libeconf/libeconf_8h.html#a7ff92b0fb657aa21220a89a0fcb59227
- https://github.com/openSUSE/libeconf/issues/81

Perhaps it would be useful do explain why the proposed order is different/better than what is currently implemented in the referenced projects.

Reply:
clerical error